packages/sdk/src/sync-agent/members/members.ts

```diff
@@ -0,0 +1,121 @@
+import type { RiverConnection } from '../river-connection/riverConnection'
+import { Member } from './models/member'
+import { isUserId } from '../../id'
+import { Myself } from './models/myself'
+
+type MembersModel = {
+    id: string
+    userIds: string[]
+    initialized: boolean
+}
+
+@persistedObservable({ tableName: 'members' })
+export class Members extends PersistedObservable<MembersModel> {
+    private members: Record<string, Member>
+    private _myself?: Myself // better naming? me, myself, myProfile?
+    constructor(streamId: string, private riverConnection: RiverConnection, store: Store) {
+        super({ id: streamId, userIds: [], initialized: false }, store)
+        this.members = {}
+    }
+
+    protected override async onLoaded() {
+        this.riverConnection.registerView((client) => {
+            if (
+                client.streams.has(this.data.id) &&
+                client.streams.get(this.data.id)?.view.isInitialized
+            ) {
+                this.onStreamInitialized(this.data.id)
+            }
+            client.on('streamInitialized', this.onStreamInitialized)
+            client.on('streamNewUserJoined', this.onMemberJoin)
+            client.on('streamNewUserInvited', this.onMemberJoin)
+            client.on('streamUserLeft', this.onMemberLeave)
+
+            return () => {
+                client.off('streamInitialized', this.onStreamInitialized)
+                client.off('streamNewUserJoined', this.onMemberJoin)
+                client.off('streamNewUserInvited', this.onMemberJoin)
+                client.off('streamUserLeft', this.onMemberLeave)
+            }
+        })
+    }
+
+    // Lazy loading the myself object, so we dont create unneeded e.g: if we're not in the stream yet
+    // but we create it if we want to access it
+    get myself() {
+        if (this._myself) return this._myself
+        this._myself = new Myself(this.data.id, this.riverConnection, this.store)
+        return this._myself
+    }
+
+    get(userId: string) {
+        check(isUserId(userId), 'invalid user id')
+        if (userId === this.riverConnection.userId) {
+            return this.myself
+        }
+        // Its possible to get a member that its not in the userIds array, if the user left the stream for example
+        // We can get a member that left, to get the last snapshot of the member
+        if (!this.members[userId]) {
+            this.members[userId] = new Member(
+                userId,
+                this.data.id,
+                this.riverConnection,
+                this.store,
+            )
+        }
+        return this.members[userId]
+    }
+
+    isUsernameAvailable(username: string): boolean {
+        const streamId = this.data.id
+        const streamView = this.riverConnection.client?.stream(streamId)?.view
+        check(isDefined(streamView), 'stream not found')
+        return streamView.getUserMetadata().usernames.cleartextUsernameAvailable(username)
+    }
+
```

Comment:
I renamed getUserMetadata to getMemberMetadata - if you kill all local dev and run ./scripts/yarn-clean.sh you should see the error